reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportInstanceIT.kt

```diff
@@ -0,0 +1,290 @@
+        val reportDefinitionRequest = constructReportDefinitionRequest()
+        val reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val onDemandRequest = """
+            {}
+        """.trimIndent()
+        val onDemandResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/on_demand/$reportDefinitionId",
+            onDemandRequest,
+            200
+        )
+
+        val updateStatus = """
+            {
+                "status":"Success",
+                "statusText":"Operation completed"
+            }
+        """.trimIndent()
+        val reportInstance = onDemandResponse.get("reportInstance").asJsonObject
+        val reportInstanceId = reportInstance.get("id").asString
+        val reportDefinitionUpdateResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/instance/$reportInstanceId",
+            updateStatus,
+            200
+        )
+        Assert.assertNotNull("report definition should be updated", reportDefinitionUpdateResponse)
+        Assert.assertEquals(
+            reportInstanceId,
+            reportDefinitionUpdateResponse.get("reportInstanceId").asString
+        )
+        val reportInstanceInfo = executeRequest(
+            RestRequest.Method.GET.name,
+            "$BASE_REPORTS_URI/instance/$reportInstanceId",
+            "",
+            200
+        )
+        val updatedReportInstance = reportInstanceInfo.get("reportInstance").asJsonObject
+        val updatedReportInstanceStatus = updatedReportInstance.get("status").asString
+        val updatedReportInstanceText = updatedReportInstance.get("statusText").asString
+        Assert.assertEquals(
+            "Success",
+            updatedReportInstanceStatus
+        )
+        Assert.assertEquals(
+            "Operation completed",
+            updatedReportInstanceText
+        )
+    }
+
+    fun `test update report instance status with invalid instance id`() {
+        val reportDefinitionRequest = constructReportDefinitionRequest()
+        val reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val updateStatus = """
+            {
+                "status":"Success",
```

Comment:
probably we can also test status=failed case somewhere

Reply:
Sure, will add case